ACKReactiveExtensions/Marshal/MarshalMapping.swift

```diff
@@ -0,0 +1,128 @@
+    }
+    
+    /**
+     * Map value as `Unmarshaling` object
+     *
+     * - parameter key: If your objects are contained within dictionary pass the key here
+     */
+    public func mapResponseMarshal<Model>(for key: KeyType? = nil) -> Signal<[Model], Error> where Model: Unmarshaling {
+        return attemptMap { json in
+            Result {
+                if let key = key, let marshaledJSON = json as? MarshaledObject {
+                    return try marshaledJSON.value(for: key)
+                }
+                else if let marshaledArray = json as? [MarshaledObject] {
+                    let dummyKey = "dummyKey"
+                    return try [dummyKey: marshaledArray].value(for: dummyKey)
+                }
+                else {
+                    assertionFailure("json isn't any of the known MarshaledObject types (Dictionary or Array)")
+                    throw NSError(domain: "", code: 0, userInfo: nil)
+                }
+                }
+                .mapError { Error.createMarshalError($0) }
+        }
+    }
+    
+    /**
+     * Map value as `ValueType`
+     *
+     * - parameter key: If your objects are contained within dictionary pass the key here
+     */
+    public func mapResponseMarshal<Model>(for key: KeyType) -> Signal<Model, Error> where Model: ValueType {
+        return attemptMap { json in
+            Result {
+                guard let marshaledJSON = json as? MarshaledObject
+                    else {
+                        assertionFailure("json isn't any of the known MarshaledObject types (Dictionary or Array)")
+                        throw NSError(domain: "", code: 0, userInfo: nil)
+                }
+                return try marshaledJSON.value(for: key)
+                }
+                .mapError { Error.createMarshalError($0) }
+        }
+    }
+}
+
+extension SignalProducerProtocol where Value == Any, Error: MarshalErrorCreatable {
+    /**
+     * Map value as `Unmarshaling` object
+     *
+     * - parameter key: If your objects are contained within dictionary pass the key here
+     */
+    public func mapResponseMarshal<Model>(for key: KeyType? = nil) -> SignalProducer<Model, Error> where Model: Unmarshaling {
+        return lift { $0.mapResponseMarshal(for: key) }
+    }
+    
+    /**
+     * Map value as `Unmarshaling` object
+     *
+     * - parameter key: If your objects are contained within dictionary pass the key here
+     */
+    public func mapResponseMarshal<Model>(for key: KeyType? = nil) -> SignalProducer<[Model], Error> where Model: Unmarshaling {
+        return lift { $0.mapResponseMarshal(for: key) }
+    }
+    
+    /**
+     * Map value as `ValueType`
+     *
+     * - parameter key: If your objects are contained within dictionary pass the key here
+     */
+    public func mapResponseMarshal<Model>(for key: KeyType) -> SignalProducer<Model, Error> where Model: ValueType {
+        return lift { $0.mapResponseMarshal(for: key) }
+    }
+}
```

Comment:
You mean something like this? https://github.com/AckeeCZ/ACKReactiveExtensions/blob/8bb0cc8dca4eb357b302234e78d01e6361ae2077/Example/Tests/MarshalMappingTests.swift